pvlib/pvsystem.py

```diff
@@ -1164,16 +1165,16 @@ def get_iam(self, aoi, iam_model='physical'):
             if `iam_model` is not a valid model name.
         """
         model = iam_model.lower()
-        if model in ['ashrae', 'physical', 'martin_ruiz']:
-            param_names = iam._IAM_MODEL_PARAMS[model]
-            kwargs = _build_kwargs(param_names, self.module_parameters)
-            func = getattr(iam, model)
+        if model in ['ashrae', 'physical', 'martin_ruiz', 'interp']:
+            func = getattr(iam, model)  # get function at pvlib.iam
+            # get all parameters from function signature to retrieve them from
+            # module_parameters if present
+            params = set(inspect.signature(func).parameters.keys())
+            params.discard('aoi')  # exclude aoi so it can't be repeated
+            kwargs = _build_kwargs(params, self.module_parameters)
```

Comment:
Makes me wonder if the IAM parameters should have been tracked in their own separate data structure instead of having to be extracted from `module_parameters`.  (no action needed in this PR; just thinking out loud)

Reply:
I think they only reason they were in `module_parameters` is because the old SAPM database includes them. That's not a reason to keep this as they are, in my view.